src/config.rs

```diff
@@ -63,17 +77,49 @@ pub struct Config {
     pub config_file_path: Option<PathBuf>,
 
     pub rule_configurations: Vec<RuleConfiguration>,
+
+    #[allow(clippy::type_complexity)]
+    #[builder(setter(skip))]
+    rules_by_plugin_prefixed_name: OnceLock<HashMap<String, (Arc<dyn Rule>, Option<PluginIndex>)>>,
 }
 
 impl Config {
-    fn get_active_rules(&self) -> Vec<Arc<dyn Rule>> {
+    fn get_rules_by_plugin_prefixed_name(
```

Comment:
A lot of these things may be questionably factored - there's a core-seeming question of "are rules "self-aware" of which plugin they belong to?"

I went with "no" as the answer to that (because it seemed like that would mean eg exposing a method on `Rule` that returns a `Plugin` reference or something, but then I'm not sure how to picture that working/getting "wired up" on the plugin-/rule-authoring side)

But so then at that point we need to keep track of the relationship between rules and the (optional) plugin they "came from" ourselves